### PR TITLE
[WIP] Port redmine issues

### PR DESCRIPTION
**Pull request by D4N**
Please **do not merge** this yet!

This is the port of the issues 426-800 from redmine from `bugfixes-test.sh` to the new python test suite. That's about half of the redmine issues, the other half still has to be ported. I would like to port them all in one big PR so that we can check that we have the same test coverage.

The branch of this PR is in the main exiv2 repository, so everyone with write access to exiv2 can push to this branch too. Please do **not** `push -f` here! I'll clean the git history up before we merge this.

I'll ask for explicit approvals later on, so please do not approve this. Approving this means "I have verified that we have exactly the same test coverage with the new test suite"

### CODE REVIEW

**Comment by D4N** [comment]
Actually I wanted to make this a "all-in-one-sweep" pr, but small steps won't hurt. Let me clean up the git history a bit (there are some useless commits from me left over) and rebase this on master once I have created the VM pull request (as that should also fix the CentOS build on gitlab).

**Comment by piponazo** [comment]
I was just my point of view, feel free to do as you please :wink:

**Comment by D4N** [comment]
@piponazo Would it be ok with you if I squased c5f2993, 5602841 and 71008bf? At least my commit from the 3 is useless, and the backslashes look like a mistake to me. I'd squash them into one commit with the message "Fix issues with python strings & literals" and would try to commit it in your name.

**Comment by piponazo** [comment]
Sure, go ahead!

**Comment by D4N** [comment]
Done. The gitlab build (actually only the CentOS build) will still fail as #320 is not merged yet.

**Comment by piponazo** [comment]
@D4N Should not we remove the tests from the bash test-suite? Otherwise we will have duplicated tests

File: ci/run.sh [hunk 1]
```diff
@@ -21,6 +21,3 @@ make tests
 make install
 cd bin
 ./unit_tests
-
-cd ../../tests/
-python3 runner.py
```
Comment (piponazo) [inline ci/run.sh]: Is this removing the execution of tests from travis?

Comment (D4N) [reply ci/run.sh]: No, they are run by `make tests`. @clanmills added them to the test makefile as `newtests`

**Comment by D4N** [comment]
> @D4N Should not we remove the tests from the bash test-suite? Otherwise we will have duplicated tests

I'd rather not, as we do not have the same test coverage yet. I would like to port all the issues first, then compute the test coverage, deactivate the bash tests and then recheck the test coverage.

**Review by piponazo — APPROVED** [review]

**Pull request merged**